execution_engine_testing/test_support/src/wasm_test_builder.rs

```diff
@@ -298,11 +307,65 @@ impl LmdbWasmTestBuilder {
         path.push(GLOBAL_STATE_DIR);
         path
     }
+
+    /// Execute and commit transforms from an ExecuteRequest into a scratch global state.
+    /// You MUST call scratch_flush to flush these changes to LmdbGlobalState.
+    pub fn scratch_exec(&mut self, mut exec_request: ExecuteRequest) -> &mut Self {
```

Comment:
Since we use `ScratchGlobalState` in the contract runtime to execute all deploys, it should be the default (and the only option) in the tests as well - we should modify the `WasmTestBuilder::exec` to use `ScratchGlobalState`.

Reply:
I agree that `WasmTestBuilder` needs some work, but it's not true today that all `exec` calls go through `ScratchGlobalState`.

Reply:
All calls to `operations::execute_finalized_block` use `ScratchGlobalState`. Which calls do not use it (but use a non-`ScratchGlobalState`)?

Reply:
Step is called through an `ExecuteRequest`, and uses `WasmTestBuilder::exec`. `execute_finalized_block` exists in the node and we don't call it directly.

Reply:
I didn't realize that `Step` is executed via `WasmTestBuilder::exec`.